oauthproxy.go

```diff
@@ -658,6 +656,8 @@ func (p *OAuthProxy) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
 		p.AuthenticateOnly(rw, req)
 	case path == p.UserInfoPath:
 		p.UserInfo(rw, req)
+	case p.IsWhitelistedRequest(req):
```

Comment:
I think it would be better to move the AuthOnlyPath case rather than this case, this change currently affects a lot of the code flow as a request to SignInPath is no longer skipped if the request is whitelisted. While this may be desirable, we should have a discussion about this first.